capstone-frontend/src/components/ClubSearchList.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { CardDeck, Spinner } from 'react-bootstrap';
+import ClubGridItem from './ClubGridItem';
+
+import '../styles/Groups.css';
+
+export class ClubSearchList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clubs: [],
+      loading: true,
+    }
+  }
+
+  getData = async () => {
+    // Get clubs by searchQuery
+    let clubs = await fetch(`/api/clubSearch?searchTerm=${this.props.searchQuery}`)
+      .then(response => response.json())
+      .catch(function (err) {
+        alert(err)
+      });
+
+    // Get clubs by checking to see if any clubs are reading the books found in Google Books API search
+    await Promise.all(this.props.books.map(async (book) => {
+      const clubsWithBook = await fetch(`/api/clubs?gbookID=${book.id}`)
```

Comment:
The `await` here makes it so that each network request only goes out once the previous one has completed the round trip. You should be able to omit `await` and modify the `clubs` array in your `.then` callback to do the processing in parallel, same for the requests generated inside loops below. In general, `await` is only needed when the subsequent steps need access to the returned data, otherwise you're unnecessarily throttling yourself.

Reply:
This makes sense! I was running into issues with there being no return value from my fetch when I removed the internal async await.

Met offline to discuss and resolve it and learned more about how Promises work and how to properly return the inside promise.